### DISCUSSION TITLE
Allow `Step#spec` to be a generic to give a data constraint to spec handlers

### DISCUSSION THREAD

**Discussion by jmcdo29** [pactumjs/pactum#134] [Ideas]
Hey there! Been loving using Pactum for some E2E tests recently, working on a full example suite using PactumJS, NestJS, and uvu! I've noticed that there's times I'd like to have my named specs pass in structured data, but right now pactum types that as `any`. I was hoping you wouldn't mind a small PR to update the `step.d.ts` file to set `Step#spec` to be typed as
```ts
declare class Step {
  spec<T = any>(name?: string, data?: T): StepSpec;
  clean(name?: string, data?: any): CleanStep;
  toss(): Promise<void>;
}
```
This shouldn't bring in any breaking changes, as the type by default is still `any`, but now I could do something like

```ts
NeighborSuite('Add another neighborhood', async ({ neighborE2E }) => {
  const anotherNeighborhoodStep = neighborE2E.step('Add Another Neighborhood');
  await anotherNeighborhoodStep
    .spec<{ key: string; overrides: Record<string, string> }>( // <---- the generic right here gives stricter typings for the second parameter
      NeighborhoodConstants.specs.add,
      {
        key: 'NewNeighborhood',
        overrides: {
          name: 'NewNeighborhood',
          location: 'N',
        },
      },
    )
    .toss(),
    anotherNeighborhoodStep.clean(NeighborhoodConstants.specs.delete, {
      id: `$S{NewNeighborhood${NeighborhoodConstants.keys.id}}`,
    });
});
```
And have stricter typings for the data passed into the `spec()`

**Comment by ASaiAnudeep** [comment]
Hey @jmcdo29, thank you for the suggestion. If it's not going to break anything, we can extend the same behaviour to the rest of the library.

Please raise a PR with the changes.

**Reply by jmcdo29** [reply]
Pull request issued: https://github.com/pactumjs/pactum/pull/136
